### DISCUSSION TITLE
using Web Speech API on linux

### DISCUSSION THREAD

**Discussion by social-booster-ba** [tauri-apps/tauri#8784] [General]
Hi there! Hope you are doing great. I have developed a Tauri app with Next js that uses Web Speech API for text to speech and speech recognition. The windows bundle works perfectly but when I try it on linux the window.speechSynthesis is undefined. Does anyone know what would the best way to approach this would be? My goal is to use the app on raspberry.

Ideas I have looked for:
-Web Speech API alternatives for react (not found)
-Running the windows bundle with WineHQ (fails to install webview2)
-Installing WebView2 with WineHQ (instalation fails)
-My last resort would be installing windows arm but I understand audio drivers do not work properly.

I will very much appreciate any help you can give me. Thank you all!

**Comment by Pietagorh** [comment]
hey, @social-booster-ba i had this problem as well
If you want to use speechSynthesis from the WebSpeechAPI on Linux you apparently need to build WebKitGTK yourself, and add `-DUSE_SPIEL=ON -DUSE_FLITE=OFF` to the cmake command :)
it's not officially fully supported tho and each user has to do it since it's an os dependency :/
speech recognition isn't supported at all tho, and it'll take some time to get it added

best option is integrating alternatives for both, like whisper.cpp for stt and piper for tts
